src/Task/FileSystem/CopyDir.php

```diff
@@ -32,7 +32,9 @@ protected function copyDir($src, $dst)
         if (false === $dir) {
             throw new TaskException($this, "Cannot open source directory '" . $src . "'");
         }
-        @mkdir($dst);
+        if (!is_dir($dst)) {
+            mkdir($dst, 0777, true);
```

Comment:
It's not like that standard php is secure at all. I think especially in the case of dealing with assets that are usually going into a *public* webroot it is critical to **not** set them to 777. This will make it very easy to add malicious files and code into files.